File: templates/python/control_mpc_phase_portrait.py
"""control_mpc_phase_portrait: MPC 控制进阶相平面画像（phase-plane 模式，合成数据）."""
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', '_utils', 'python'))
from generated_patterns import make_template_figure


def make_figure(title='advanced MPC control: phase portrait'):
    return make_template_figure('phase_plane', seed=1611, title=title, domain='advanced MPC control', topic='phase portrait')


if __name__ == '__main__':
    fig = make_figure()
    fig.savefig(__file__.replace('.py', '.png'), dpi=150)

File: templates/matlab/control_mpc_phase_portrait.m
function fig = control_mpc_phase_portrait()
    addpath(fullfile(fileparts(mfilename('fullpath')),'..','..','_utils','matlab'));
    fig = generated_template_figure('phase_plane', 1611, 'advanced MPC control: phase portrait', 'advanced MPC control', 'phase portrait');
end
